wallet/src/Cardano/Wallet/WalletLayer/Kernel/Internal.hs

```diff
@@ -110,18 +112,26 @@ importWallet :: MonadIO m
              => Kernel.PassiveWallet
              -> WalletImport
              -> m (Either ImportWalletError Wallet)
-importWallet pw WalletImport{..} = liftIO $ do
-    secretE <- try $ Keystore.readWalletSecret wiFilePath
-    case secretE of
-         Left e ->
-             if isDoesNotExistError e
-                 then return (Left $ ImportWalletFileNotFound wiFilePath)
-                 else throwM e
-         Right mbEsk -> do
-             case mbEsk of
-                 Nothing  -> return (Left $ ImportWalletNoWalletFoundInBackup wiFilePath)
-                 Just esk -> do
-                     res <- liftIO $ createWallet pw (ImportWalletFromESK esk wiSpendingPassword)
-                     return $ case res of
-                          Left e               -> Left (ImportWalletCreationFailed e)
-                          Right importedWallet -> Right importedWallet
+importWallet pw WalletImport{wiSpendingPassword,wiFilePath,wiRawSecret} = liftIO $ do
+    eskE <- case (wiFilePath, wiRawSecret) of
+      (Nothing, Nothing) -> return $ Left ImportWalletMissingField
+      (Just _, Just _) -> return $ Left ImportWalletMissingField
```

Comment:
Not really missing though :/

Reply:
either missing, or too many, but i didnt feel like adding yet another error type, and all the build instances everywhere

and then during testing, i discovered it cant actually return these errors to the client, its always an empty error 500, and the exception is printed to stdout! (bypassing logging!!!)

Reply:
it would also be better to make the api accept an `Either`, but then its no longer backwards compatible, and things have to be fixed elsewhere

Reply:
> i didnt feel like adding yet another error type

I hear you, but that's how we get the frontend to report weird error to us in the end. If we don't report errors correctly, how can we expect API consumer to then, report them correctly as well?

Reply:
>  its always an empty error 500, and the exception is printed to stdout! (bypassing logging!!!)

That is... not cool at all and not normal. You should by the way make sure that errors are mapped into the "current / old errors" from the API, here:

https://github.com/input-output-hk/cardano-sl/blob/ca76e458b119337796e4991ffe9716c7d2f4c991/wallet/src/Cardano/Wallet/API/V1/ReifyWalletError.hs